collect_app/src/main/java/org/odk/collect/android/widgets/PrinterWidget.kt

```diff
@@ -0,0 +1,62 @@
+package org.odk.collect.android.widgets
+
+import android.app.Activity
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.button.MaterialButton
+import org.javarosa.core.model.data.IAnswerData
+import org.javarosa.form.api.FormEntryPrompt
+import org.odk.collect.android.R
+import org.odk.collect.android.formentry.questions.QuestionDetails
+import org.odk.collect.android.utilities.QuestionMediaManager
+import org.odk.collect.android.widgets.interfaces.Printer
+import org.odk.collect.material.MaterialProgressDialogFragment
+
+class PrinterWidget(
+    context: Context,
+    questionDetails: QuestionDetails,
+    private val printer: Printer,
+    private val questionMediaManager: QuestionMediaManager
+) : QuestionWidget(context, questionDetails) {
+
+    init {
+        render()
+
+        MaterialProgressDialogFragment.showOn(
```

Comment:
This observation should move up to `FormFillingActivity`. Currently, if we had multiple `PrinterWidget` instances on screen, we'd end up making multiple observations that show multiple dialogs (as we call `showOn` for every view creation).

Reply:
but the dialog will be shown only when `isLoading` is true and it's true when you click the print button. You are not able to click multiple buttons at once so there will never be multiple dialogs displayed right?

Reply:
That's fair, but I still think it makes more sense for the observation to live at the Activity level that than in view creation.

Reply:
Ok, done.